monitoring/snippets/metric_create.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2015 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+package snippets
+
+// [START monitoring_create_metric]
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	monitoring "cloud.google.com/go/monitoring/apiv3"
+	"google.golang.org/genproto/googleapis/api/label"
+	"google.golang.org/genproto/googleapis/api/metric"
+	metricpb "google.golang.org/genproto/googleapis/api/metric"
+	monitoringpb "google.golang.org/genproto/googleapis/monitoring/v3"
+)
+
+// createCustomMetric creates a custom metric specified by the metric type.
+func createCustomMetric(w io.Writer, projectID, metricType string) (*metricpb.MetricDescriptor, error) {
+	ctx := context.Background()
+	c, err := monitoring.NewMetricClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+	md := &metric.MetricDescriptor{
+		Name: "Custom Metric",
+		Type: metricType,
+		Labels: []*label.LabelDescriptor{{
+			Key:         "environment",
+			ValueType:   label.LabelDescriptor_STRING,
+			Description: "An arbitrary measurement",
+		}},
+		MetricKind:  metric.MetricDescriptor_GAUGE,
+		ValueType:   metric.MetricDescriptor_INT64,
+		Unit:        "s",
+		Description: "An arbitrary measurement",
+		DisplayName: "Custom Metric",
+	}
+	req := &monitoringpb.CreateMetricDescriptorRequest{
+		Name:             "projects/" + projectID,
+		MetricDescriptor: md,
+	}
+	metric, err := c.CreateMetricDescriptor(ctx, req)
```

Comment:
nit: `metric` collides with the package name. Maybe choose something else? (here and elsewhere)

Reply:
Done.